src/experimental/ProbabilisticGraphicalModels/bayesian_network.jl

```diff
@@ -29,7 +30,8 @@ function BayesianNetwork{V}() where {V}
         SimpleDiGraph{Int}(), # by default, vertex ids are integers
         V[],
         Dict{V,Int}(),
-        Dict{V,Any}(),
+        (;),    # Empty NamedTuple for evaluation_env
```

Comment:
Because `BayesianNetwork` is immutable, although we could use `BangBang.setfield!!` to set it later, maybe a better way to define this constructor is to take `evaluation_env` as an argument.